platform/security/pdp/security-pdp-xacmlrealm/src/main/java/ddf/security/pdp/xacml/processor/BalanaPdp.java

```diff
@@ -363,26 +352,34 @@ private String marshal(RequestType xacmlRequestType) throws PdpException {
     /**
      * Unmarshalls the XACML response.
      *
-     * @param xacmlResponse
-     *            The XACML response with all namespaces and namespace prefixes added.
+     * @param xacmlResponse The XACML response with all namespaces and namespace prefixes added.
      * @return The XACML response.
      * @throws PdpException
      */
+    @SuppressWarnings("unchecked")
     private ResponseType unmarshal(DOMResult xacmlResponse) throws PdpException {
+        List<String> ctxPath = ImmutableList.of(ResponseType.class.getPackage()
+                .getName());
+
+        if (null == parser) {
+            throw new IllegalStateException("XMLParser must be configured.");
+        }
+
+        ParserConfigurator configurator = parser.configureParser(ctxPath,
```

Comment:
If you moved these lines inside the `try` block, you wouldn't need to declare the `xacmlResponseTypeElement` as null prior to unmarshalling.